docs/dg/dev/upgrade-and-migrate/upgrade-to-angular-17.md

```diff
@@ -0,0 +1,474 @@
+
+This document provides instructions for upgrading Angular to version 17 in your Spryker project.
+
+## Overview
+
+Our current version of Angular is v15.
+Angular v15 should be deprecated on 2024-05-18, according to the [policy](https://angular.io/guide/releases#support-policy-and-schedule).
+The current stable version of Angular is v17.
+
+
+Upgrade to the latest major version of Angular to get the most recent bug fixes and security updates. Additionally, this will lead to optimization in both runtime performance and tooling.
+
+Keep in mind that updating to Angular v17 results in incompatibility with older versions of Angular. Therefore, a major release is necessary for these modules:
+
+- `AgentSecurityMerchantPortalGui`
+- `AgentDashboardMerchantPortalGui`
+- `DashboardMerchantPortalGui`
+- `GuiTable`
+- `MerchantProfileMerchantPortalGui`
+- `ProductMerchantPortalGui`
+- `ProductOfferMerchantPortalGui`
+- `SalesMerchantPortalGui`
+- `SecurityMerchantPortalGui`
+- `UserMerchantPortalGui`
+- `ZedUi`
+
+*Estimated migration time: 2h*
+
+### 1) Update modules
+
+1. Upgrade modules to the new version:
+
+The marketplace modules must correspond to the following versions:
+
+| NAME                                        | VERSION   |
+| ------------------------------------------- | --------- |
+| AgentSecurityMerchantPortalGui              | >= 2.0.0  |
+| AgentDashboardMerchantPortalGui             | >= 2.0.0  |
+| DashboardMerchantPortalGui                  | >= 2.0.0  |
+| GuiTable                                    | >= 2.0.0  |
+| MerchantProfileMerchantPortalGui            | >= 2.0.0  |
+| ProductMerchantPortalGui                    | >= 3.0.0  |
+| ProductOfferMerchantPortalGui               | >= 2.0.0  |
+| SalesMerchantPortalGui                      | >= 2.0.0  |
+| SecurityMerchantPortalGui                   | >= 2.0.0  |
+| UserMerchantPortalGui                       | >= 2.0.0  |
+| ZedUi                                       | >= 2.0.0  |
+
+If not, update module versions manually or by using the following command:
+
+```bash
+composer update spryker/agent-dashboard-merchant-portal-gui spryker/agent-security-merchant-portal-gui spryker/dashboard-merchant-portal-gui spryker/gui-table spryker/merchant-profile-merchant-portal-gui spryker/product-merchant-portal-gui spryker/product-offer-merchant-portal-gui spryker/sales-merchant-portal-gui spryker/security-merchant-portal-gui spryker/user-merchant-portal-gui spryker/zed-ui
+```
+
+2. Regenerate the data transfer object:
+
+```bash
+console transfer:generate
+```
+
+## 2) Update Stylelint and css-loader
+
+Before starting the migration, make sure that stylelint and css-loader were updated by this doc: [Stylelint migration guide](/docs/dg/dev/upgrade-to-stylelint-16-and-css-loader-6.html) is required.
+
+## 3) Update npm dependencies
+
+1. In `package.json`, do the following:
+
+   1. Update or add the following dependencies and change commands:
```

Comment:
```suggestion
   1. Update or add the following dependencies and change npm scripts:
```

Reply:
Maybe it makes sense to separate it into 3 separate sections, as long as for now it looks like a complete content for package.json. so something like this looks a bit better:

```
"scripts": {
    ...
    "mp:build": "nx build",
    "mp:build:watch": "nx build --watch",
    "mp:build:production": "nx build --configuration production",
    "mp:lint": "nx lint --no-eslintrc",
    "mp:test": "nx test",
    ...
},
```